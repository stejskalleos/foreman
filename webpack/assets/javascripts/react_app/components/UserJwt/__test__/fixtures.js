import { noop } from '../../../common/helpers';
import { STATUS } from '../../../constants';

export const formProps = {
  handleSubmit: noop,
};

const token = 'token23';

export const fieldOk = {
  apiStatus: STATUS.RESOLVED,
  token,
};

export const fieldError = {
  apiStatus: STATUS.ERROR,
  token,
};

export const fieldPending = {
  apiStatus: STATUS.PENDING,
  token,
};
