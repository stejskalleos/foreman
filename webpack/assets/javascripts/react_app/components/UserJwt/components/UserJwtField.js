import React from 'react';
import { Grid, Alert, LoadingState } from 'patternfly-react';
import PropTypes from 'prop-types';

import { translate as __ } from '../../../common/I18n';
import { STATUS } from '../../../constants';
import CopyToClipboard from '../../common/CopyToClipboard/CopyToClipboard';

const UserJwtField = ({ apiStatus, token }) => {
  if (apiStatus === STATUS.ERROR) {
    return (
      <Grid.Row>
        <Grid.Col md={12}>
          <Alert>
            {__(
              'There was an error while creating the JWT token, try refreshing the page.'
            )}
          </Alert>
        </Grid.Col>
      </Grid.Row>
    );
  }

  return (
    <LoadingState loading={apiStatus === STATUS.PENDING}>
      {token && (
        <React.Fragment>
          <Grid.Row>
            <Grid.Col md={12}>
              <div className="user-jwt-field">
                <p>
                  {__(
                    'Following JWT can be used for the API authorization, like this:'
                  )}
                  <br />
                  <code>
                    curl -X GET &apos;
                    {`${window.location.hostname}/api/v2/hosts`}
                    &apos;
                    -H&nbsp;&apos;Authorization:&nbsp;Bearer&nbsp;your-token&apos;
                  </code>
                </p>
              </div>
            </Grid.Col>
          </Grid.Row>
          <Grid.Row>
            <Grid.Col md={4}>
              <strong>{__('Generated token:')}</strong>
              <CopyToClipboard valueToCopy={token} />
            </Grid.Col>
          </Grid.Row>
        </React.Fragment>
      )}
    </LoadingState>
  );
};

UserJwtField.propTypes = {
  apiStatus: PropTypes.string,
  token: PropTypes.string,
};

UserJwtField.defaultProps = {
  apiStatus: null,
  token: null,
};

export default UserJwtField;
