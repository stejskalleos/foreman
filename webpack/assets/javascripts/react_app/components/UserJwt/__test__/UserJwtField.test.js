import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';

import UserJwtField from '../components/UserJwtField';
import { fieldOk, fieldError, fieldPending } from './fixtures';

const fixtures = {
  ok: fieldOk,
  error: fieldError,
  pending: fieldPending,
};

describe('UserJwtField', () => {
  testComponentSnapshotsWithFixtures(UserJwtField, fixtures);
});
