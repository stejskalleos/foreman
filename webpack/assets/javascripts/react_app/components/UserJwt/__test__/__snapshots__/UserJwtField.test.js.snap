// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserJwtField error 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={12}
  >
    <Alert
      className=""
      onDismiss={null}
      type="error"
    >
      There was an error while creating the JWT token, try refreshing the page.
    </Alert>
  </Col>
</Row>
`;

exports[`UserJwtField ok 1`] = `
<LoadingState
  additionalClasses=""
  loading={false}
  loadingText="Loading"
  size="lg"
  timeout={300}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={12}
    >
      <div
        className="user-jwt-field"
      >
        <p>
          Following JWT can be used for the API authorization, like this:
          <br />
          <code>
            curl -X GET '
            localhost/api/v2/hosts
            ' -H 'Authorization: Bearer your-token'
          </code>
        </p>
      </div>
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
    >
      <strong>
        Generated token:
      </strong>
      <CopyToClipboard
        valueToCopy="token23"
      />
    </Col>
  </Row>
</LoadingState>
`;

exports[`UserJwtField pending 1`] = `
<LoadingState
  additionalClasses=""
  loading={true}
  loadingText="Loading"
  size="lg"
  timeout={300}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={12}
    >
      <div
        className="user-jwt-field"
      >
        <p>
          Following JWT can be used for the API authorization, like this:
          <br />
          <code>
            curl -X GET '
            localhost/api/v2/hosts
            ' -H 'Authorization: Bearer your-token'
          </code>
        </p>
      </div>
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
    >
      <strong>
        Generated token:
      </strong>
      <CopyToClipboard
        valueToCopy="token23"
      />
    </Col>
  </Row>
</LoadingState>
`;
